test/org/daisy/dotify/formatter/impl/segment/TextSegmentTest.java

```diff
@@ -0,0 +1,34 @@
+package org.daisy.dotify.formatter.impl.segment;
+
+import org.daisy.dotify.api.formatter.TextProperties;
+import org.junit.Test;
+
+import java.util.Optional;
+
+import static org.junit.Assert.*;
+
+public class TextSegmentTest {
+    @Test
+   public void testGetLocaleReturnEmptyOptionalIfLocaleIsNull() {
+        // Test that verifies that getLocale does not crash if locale is null.
+        TextProperties.Builder builder = new TextProperties.Builder(null);
+        TextProperties tp = builder.build();
+        String chars = "";
+        TextSegment ts = new TextSegment(chars, tp, false);
+        Optional <String> locale = ts.getLocale();
+        assertEquals(Optional.empty(), locale);
+    }
+
+    @Test
+    public void testVerifyComparisonWIthOptionals() {
```

Comment:
These test is just to verify the behaviour when comparing the returned value from getLocal.